File: app/src/main/java/com/esoxjem/movieguide/sorting/SortingDialogPresenter.java
package com.esoxjem.movieguide.sorting;

/**
 * @author arun
 */
public class SortingDialogPresenter implements ISortingDialogPresenter
{
    private ISortingDialogView view;
    private ISortingDialogInteractor sortingDialogInteractor;

    public SortingDialogPresenter(ISortingDialogInteractor interactor)
    {
        sortingDialogInteractor = interactor;
    }

    @Override
    public void setView(ISortingDialogView view)
    {
        this.view = view;
    }

    @Override
    public void destroy()
    {
        view = null;
    }

    @Override
    public void setLastSavedOption()
    {
        if (isViewAttached())
        {
            int selectedOption = sortingDialogInteractor.getSelectedSortingOption();

            if (selectedOption == SortType.MOST_POPULAR.getValue())
            {
                view.setPopularChecked();
            } else if (selectedOption == SortType.HIGHEST_RATED.getValue())
            {
                view.setHighestRatedChecked();
            } else
            {
                view.setFavoritesChecked();
            }
        }
    }

    private boolean isViewAttached()
    {
        return view != null;
    }

    @Override
    public void onPopularMoviesSelected()
    {
        if (isViewAttached())
        {
            sortingDialogInteractor.setSortingOption(SortType.MOST_POPULAR);
            view.dismissDialog();
        }
    }

    @Override
    public void onHighestRatedMoviesSelected()
    {
        if (isViewAttached())
        {
            sortingDialogInteractor.setSortingOption(SortType.HIGHEST_RATED);
            view.dismissDialog();
        }
    }

    @Override
    public void onFavoritesSelected()
    {
        if (isViewAttached())
        {
            sortingDialogInteractor.setSortingOption(SortType.FAVORITES);
            view.dismissDialog();
        }
    }
}

File: app/src/main/java/com/esoxjem/movieguide/movies/sorting/SortingDialogPresenter.kt
package com.esoxjem.movieguide.movies.sorting

/**
 * @author arunsasidharan
 */
 class SortingDialogPresenter(
        private val sortingDialogInteractor: SortingContract.Interactor) :
        SortingContract.Presenter {
    private var view: SortingContract.View? = null

    override fun setView(view: SortingContract.View) {
        this.view = view
    }

    override fun destroy() {
        view = null
    }

    override fun setLastSavedOption() {
        val selectedOption = sortingDialogInteractor.getSelectedSortingOption()

        when (selectedOption) {
            SortType.MOST_POPULAR.value -> view?.setPopularChecked()
            SortType.HIGHEST_RATED.value -> view?.setHighestRatedChecked()
            SortType.NEWEST.value -> view?.setNewestChecked()
            else -> view?.setFavoritesChecked()
        }
    }

    override fun onPopularMoviesSelected() {
        sortingDialogInteractor.setSortingOption(SortType.MOST_POPULAR)
        view?.dismissDialog()
    }

    override fun onHighestRatedMoviesSelected() {
        sortingDialogInteractor.setSortingOption(SortType.HIGHEST_RATED)
        view?.dismissDialog()
    }

    override fun onNewestMoviesSelected() {
        sortingDialogInteractor.setSortingOption(SortType.NEWEST)
        view?.dismissDialog()
    }

    override fun onFavoritesSelected() {
        sortingDialogInteractor.setSortingOption(SortType.FAVORITES)
        view?.dismissDialog()
    }
}
